src/registrar/forms/domain_request_wizard.py

```diff
@@ -22,10 +22,151 @@
+        cleaned_data = super().clean()
+
+        # Do some custom error validation if the requesting entity is a suborg.
+        # Otherwise, just validate as normal.
+        suborganization = self.cleaned_data.get("sub_organization")
+        is_requesting_new_suborganization = self.cleaned_data.get("is_requesting_new_suborganization")
+
+        # Get the value of the yes/no checkbox from RequestingEntityYesNoForm.
+        # Since self.data stores this as a string, we need to convert "True" => True.
+        requesting_entity_is_suborganization = self.data.get(
+            "portfolio_requesting_entity-requesting_entity_is_suborganization"
+        )
+        if requesting_entity_is_suborganization == "True":
+            if is_requesting_new_suborganization:
+                # Validate custom suborganization fields
+                if not cleaned_data.get("requested_suborganization"):
+                    self.add_error("requested_suborganization", "Enter details for your organization name.")
+                if not cleaned_data.get("suborganization_city"):
+                    self.add_error("suborganization_city", "Enter details for your city.")
+                if not cleaned_data.get("suborganization_state_territory"):
+                    self.add_error("suborganization_state_territory", "Enter details for your state or territory.")
+            elif not suborganization:
+                self.add_error("sub_organization", "Select a suborganization.")
+
+        return cleaned_data
+
+
+class RequestingEntityYesNoForm(BaseYesNoForm):
+    """The yes/no field for the RequestingEntity form."""
+
+    # This first option will change dynamically
+    form_choices = ((False, "Current Organization"), (True, "A suborganization. (choose from list)"))
+
+    # IMPORTANT: This is tied to DomainRequest.is_requesting_new_suborganization().
+    # This is due to the from_database method on DomainRequestWizard.
+    field_name = "requesting_entity_is_suborganization"
+
+    def __init__(self, *args, **kwargs):
+        """Extend the initialization of the form from RegistrarForm __init__"""
+        super().__init__(*args, **kwargs)
+        if self.domain_request.portfolio:
+            self.form_choices = (
+                (False, self.domain_request.portfolio),
+                (True, "A suborganization. (choose from list)"),
+            )
+        self.fields[self.field_name] = self.get_typed_choice_field()
+
+    @property
+    def form_is_checked(self):
+        """
+        Determines if the requesting entity is a suborganization, or a portfolio.
+        For suborganizations, users have the ability to request a new one if the
+        desired suborg doesn't exist. We expose additional fields that denote this,
+        like `requested_suborganization`. So we also check on those.
+        """
+        # True means that the requesting entity is a suborganization,
```

Comment:
**suggestion** Key info on your return values should exist in the docstring for the function. 
example to shorten it here
"""
Determines the initial checked state of the form, returning True (checked) if the requesting entity is a suborganization,  and False if it is a portfolio. Returns None if neither condition is met.

"""